api/python/provisioner/commands/setup_provisioner.py

```diff
@@ -1299,39 +1299,49 @@ def _run(self, nodes, **kwargs):  # noqa: C901 FIXME
             ), targets=run_args.primary.minion_id
         )
 
-        if run_args.target_build:
-            logger.info("Updating target build pillar")
-            ssh_client.cmd_run(
-                (
-                    'provisioner pillar_set --fpath release.sls'
-                    f' release/target_build \'"{run_args.target_build}"\''
-                ), targets=run_args.primary.minion_id
-            )
-
-            logger.info("Get release factory version")
+        logger.info("Updating target build pillar")
+        if run_args.url_cortx:
```

Comment:
That all should checked in  __post_attr_init_. Just imagine: setup almost completed with some wrong values that we may check at the very beggining and failed then.